### PR TITLE
Refactored Services and Util in AdminPanel

### LINKED ISSUE #363: Refactoring of IApiUtil and IHttpUtil in AdminPanel
AdminPanel Services include logic of obtaining Token and API URL that should be moved and used one time per request.

### PR DESCRIPTION
**Pull request by andy-dendy-repo**
**IApiUtil and IHttpUtil interfaces simplyfied:**

- Token parameter is removed from infaces' methods
- Token obtaining logic is moved to Startup class in ConfigureServices method
- API Url is set to httpclient BaseUrl during initialization of HttpUtil in Startup class

**This resulted into:**

- No need to obtain Token in Services' constructors as it's set to IHttpUtil during initialization
- No need to obtain API Url in Services' method as it's to ser to IHttpUtil during initialization
- No need to pass Token when using IHttpUtil and IApiUtil method

### CODE REVIEW

File: CharlieBackend.AdminPanel/Controllers/AccountController.cs [hunk 1]
```diff
@@ -48,9 +44,11 @@ public ViewResult Login()
         [HttpPost]
         public async Task<IActionResult> Login(AuthenticationDto authDto)
         {
-            var httpResponseToken = await _apiUtil.SignInAsync($"{_config.Value.Urls.Api.Https}/api/accounts/auth", authDto);
+            var httpResponseToken = await _apiUtil.SignInAsync($"api/accounts/auth", authDto);
+
+            httpResponseToken = httpResponseToken.Replace("Bearer ", "");
```
Comment (mahotskuy) [inline CharlieBackend.AdminPanel/Controllers/AccountController.cs]: Use Replace on 47 line.

File: CharlieBackend.AdminPanel/Startup.cs [hunk 2]
```diff
@@ -33,7 +34,28 @@ public void ConfigureServices(IServiceCollection services)
...
+
+                if (accessToken != null)
+                {
+                    IDataProtectionProvider provider = ctx.GetService<IDataProtectionProvider>();
+
+                    IDataProtector protector = provider.CreateProtector(config.Cookies.SecureKey);
+
+                    string unprotectedToken = protector.Unprotect(accessToken);
+
+                    return new HttpUtil(config.Urls.Api.Https, unprotectedToken);
```
Comment (mahotskuy) [inline CharlieBackend.AdminPanel/Startup.cs]: You can have one constructor that takes IHttpContextAccessor and IConfiguration and then extract all data in the construction.
Then depending if there is a token add it to _client.

File: CharlieBackend.AdminPanel/Controllers/AccountController.cs (hunk 1 above)
Comment (andy-dendy-repo) [reply CharlieBackend.AdminPanel/Controllers/AccountController.cs]: Solved at this [commit](https://github.com/ita-social-projects/WhatBackend/pull/362/commits/8c1d6cc0c405f1cd3b93324b26e2a4a14e43cb96)

File: CharlieBackend.AdminPanel/Startup.cs (hunk 2 above)
Comment (andy-dendy-repo) [reply CharlieBackend.AdminPanel/Startup.cs]: Startup is [restored](https://github.com/ita-social-projects/WhatBackend/pull/362/commits/7cee80a627ae3fca48e7fe2124ab075e3320dde3).
Added HttpUtil [constructor](https://github.com/ita-social-projects/WhatBackend/pull/362/commits/bde6dbbe8334e2be6a669da9eaab55e50e00e42d).

**Review by filinskyi — APPROVED** [review]

**Review by CptHouston — APPROVED** [review]

**Review by zelya2021 — APPROVED** [review]

**Review by gingerbread-monster — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

**Review by mahotskuy — APPROVED** [review]

**Pull request merged**